ptf/saiswitch.py

```diff
@@ -0,0 +1,1776 @@
+# Copyright 2021-present Intel Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Thrift SAI interface Switch tests
+"""
+from ipaddress import ip_address
+
+from sai_thrift.sai_headers import *
+
+from ptf.mask import Mask
+
+from sai_base_test import *
+
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(THIS_DIR, '..'))
+
+
+def generate_ip_addr(no_of_addr, ipv6=False):
+    '''
+    An IP address generator using generic way of addresses randomization
+    and ensuring address is unique
+
+    Args:
+        no_of_addr (int): number of requested IP addresses
+        ipv6 (bool): IP version indicator
+
+    Yield:
+        str: unique IP address
+    '''
+    if not ipv6:
+        ip_range = [addr for addr in
+                    ["1.0.0.1", "254.255.255.254"]]
+    else:
+        ip_range = [addr for addr in
+                    ["2001::0", "2001:0db8::ffff:ffff:ffff"]]
+
+    ip_interval = LpmDict.IpInterval(
```

Comment:
Please add "from lpm import LpmDict" import statement since lpm module was added to tests infrastructure in public PR.